### DISCUSSION TITLE
Sorting of Pivot Table rows needs to be driven by a certain column

### DISCUSSION THREAD

**Discussion by kpiforgeoliver** [apache/superset#30620] [Q&A / Help]
Currently, the only way to sort rows of a pivot table is by Alphabetically A>Z or Z>A or numerically by using one of the value columns that exist in the pivot table.  There needs to be a way to use the "sort by" functionality to sort the table by a column that is not being used in the table.  For Example a  'Sort_order' column that comes from the query, or a user defined column that calculates the sort order.  The current "sort by" field does absolutely nothing in this regard.  This is a major problem for customers that want to sort their metrics in a defined manor.  Currently the only work-around would be to add the "sort order" column as the first dimension value of the pivot table, but that become problematic when you have multiple dimensions and you need to drill into the pivot.

**Comment by rtotheb2000** [comment]
This request is similar to https://github.com/apache/superset/discussions/18469. I would also like to see an option to define a custom sort order.
